File: check_xpi.py
import zipfile
z = zipfile.ZipFile('releases/0.3.2/LifeUndo-0.3.2-firefox.xpi')
files = z.namelist()
print(f'XPI contains {len(files)} files')
print('manifest.json present:', 'manifest.json' in files)
print('popup.js present:', 'popup.js' in files)  
print('popup.html present:', 'popup.html' in files)
print('\nAll files:')
for f in sorted(files):
    print(f'  {f}')

File: check-xpi.js
const fs = require('fs');
const yauzl = require('yauzl');

const xpiPath = 'releases/0.2.3/LifeUndo-0.2.3-firefox.xpi';

console.log('Checking XPI contents for POSIX paths...\n');

yauzl.open(xpiPath, { lazyEntries: true }, (err, zipfile) => {
    if (err) {
        console.error('Error opening XPI:', err.message);
        return;
    }
    
    console.log('XPI file entries:');
    console.log('================');
    
    zipfile.readEntry();
    zipfile.on('entry', (entry) => {
        console.log(entry.fileName);
        
        // Check for backslashes (should be none)
        if (entry.fileName.includes('\\')) {
            console.log(`❌ ERROR: Found backslash in path: ${entry.fileName}`);
        } else {
            console.log(`✅ OK: ${entry.fileName}`);
        }
        
        zipfile.readEntry();
    });
    
    zipfile.on('end', () => {
        console.log('\n✅ XPI verification complete - all paths use POSIX format (/)');
    });
});


















